### PR TITLE
DPC-25 Common code

### PR DESCRIPTION
**Pull request by whuang85**
**Why**

Common code that can be used between the patient everything code and look back

**What Changed**

Describe the changes made to the code base including new features, refactoring of existing services and user visible API changes.

**Choices Made**

Added this code to dpc-queue instead of dpc-common as there are circular dependencies when added to dpc-common when used in dpc-attribution

**Tickets closed**:

Give a list of tickets closed in this PR.

**Future Work**

List any additional tickets that have either been created due to work in this PR, or existing tickets that expand upon the feature or provide additional fixes.

**Checklist**

- [x] All tests are passing via `make ci-app` (app change) and `make ci-web` (website change)
- [ ] Swagger documentation has been updated
- [ ] FHIR documentation has been updated
- [ ] Any required dpc-ops changes have a PR submitted and mentioned in this ticket
- [ ] Any manual migration steps are documented, scripts written (where applicable), and tested
- [ ] Before merging, any required dpc-ops changes have been approved and merged into master of the dpc-ops repo

### CODE REVIEW

File: dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    public Resource retrieveData(UUID organizationID, UUID providerID, List<String> patientIDs, ResourceType... resourceTypes) {
+        UUID jobID = this.queue.createJob(organizationID, providerID.toString(), patientIDs, List.of(resourceTypes), null, null);
+        Optional<List<JobQueueBatch>> optionalBatches = waitForJobToComplete(jobID, organizationID, this.queue);
+
+        if (optionalBatches.isEmpty()) {
+            throw new DataRetrievalException("Failed to retrieve data");
+        }
+
+        List<JobQueueBatch> batches = optionalBatches.get();
```
[the remark this answers is not in the document]
Comment (embh) [reply dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java:55]: I wonder if an `if`/`else` based on `optionalBatches.isPresent()` would make codeclimate happier.

File: dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    private Bundle assembleBundleFromBatches(List<JobQueueBatch> batches, List<ResourceType> resourceTypes) {
+        final Bundle bundle = new Bundle().setType(Bundle.BundleType.SEARCHSET);
+
+        batches.stream()
+                .map(JobQueueBatch::getJobQueueBatchFiles)
+                .flatMap(List::stream)
+                .filter(bf -> resourceTypes.contains(bf.getResourceType()))
+                .forEach(batchFile -> {
+                    java.nio.file.Path path = Paths.get(String.format("%s/%s.ndjson", exportPath, batchFile.getFileName()));
```
Comment (embh) [inline dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java:118]: What's the reason for not importing `java.nio.file.Path`?

File: dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        if (jobStatusSet.size() == 1 && jobStatusSet.contains(JobStatus.COMPLETED)) {
+            return batches;
+        } else if (jobStatusSet.contains(JobStatus.FAILED)) {
+            throw new DataRetrievalException("Failed to retrieve batches");
+        } else {
+            throw new DataRetrievalRetryException();
+        }
+    }
+
+    private Bundle assembleBundleFromBatches(List<JobQueueBatch> batches, List<ResourceType> resourceTypes) {
```
[the remark this answers is not in the document]
Comment (whuang85) [reply dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java:117]: Not sure why codeClimate doesn't see it being used at line 123.. I debugged just to make sure it is using it...

File: dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.stream.Collectors;
+
+public class DataService {
+
+    private IJobQueue queue;
+    private String exportPath;
+    private FhirContext fhirContext;
+    private int jobTimeoutInSeconds;
+
+    @Inject
```
[the remark this answers is not in the document]
Comment (whuang85) [reply dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java:39]: I don't quite understand what this is pertaining to

**Pull request merged**